src/web/rustdoc.rs

```diff
@@ -762,6 +771,118 @@ mod test {
         })
     }
 
+    #[test]
+    fn redirect_latest_does_not_go_to_yanked_versions() {
+        wrapper(|env| {
+            let db = env.db();
+            db.fake_release()
+                .name("dummy")
+                .version("0.1.0")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .create()
+                .unwrap();
+            db.fake_release()
+                .name("dummy")
+                .version("0.2.0")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .create()
+                .unwrap();
+            db.fake_release()
+                .name("dummy")
+                .version("0.2.1")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .cratesio_data_yanked(true)
+                .create()
+                .unwrap();
+
+            let web = env.frontend();
+            let redirect = latest_version_redirect("/dummy/0.1.0/dummy/", web)?;
+            assert_eq!(redirect, "/dummy/0.2.0/dummy/index.html");
+
+            let redirect = latest_version_redirect("/dummy/0.2.1/dummy/", web)?;
+            assert_eq!(redirect, "/dummy/0.2.0/dummy/index.html");
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn redirect_latest_with_all_yanked() {
+        wrapper(|env| {
+            let db = env.db();
+            db.fake_release()
+                .name("dummy")
+                .version("0.1.0")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .cratesio_data_yanked(true)
+                .create()
+                .unwrap();
+            db.fake_release()
+                .name("dummy")
+                .version("0.2.0")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .cratesio_data_yanked(true)
+                .create()
+                .unwrap();
+            db.fake_release()
+                .name("dummy")
+                .version("0.2.1")
+                .rustdoc_file("dummy/index.html", b"lah")
+                .cratesio_data_yanked(true)
+                .create()
+                .unwrap();
+
+            let web = env.frontend();
+            let redirect = latest_version_redirect("/dummy/0.1.0/dummy/", web)?;
+            assert_eq!(redirect, "/dummy/0.2.1/dummy/index.html");
+
+            let redirect = latest_version_redirect("/dummy/0.2.0/dummy/", web)?;
+            assert_eq!(redirect, "/dummy/0.2.1/dummy/index.html");
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn yanked_release_shows_warning_in_nav() {
+        fn has_yanked_warning(path: &str, web: &TestFrontend) -> Result<bool, failure::Error> {
+            use html5ever::tendril::TendrilSink;
+            assert_success(path, web)?;
+            let data = web.get(path).send()?.text()?;
+            Ok(kuchiki::parse_html()
+                .one(data)
+                .select("form ul li .warn")
```

Comment:
`get_platform_links` is pretty similar to this, do you think it's worth factoring out a `html_select()` function?